jupyter-widgets-schema/v2/state.schema.json

```diff
@@ -0,0 +1,64 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "description": "Jupyter Interactive Widget State JSON schema.",
+    "type": "object",
+    "properties" : {
+        "version_major" : {
+            "description": "Format version (major)",
+            "type": "number",
+            "minimum": 2,
+            "maximum": 2
+        },
+        "version_minor" : {
+            "description": "Format version (minor)",
+            "type": "number"
+        },
+        "state": {
+            "description": "Model State for All Widget Models",
+            "type": "object",
+            "additionalProperties" : {
+                "type": "object",
+                "properties": {
+                    "model_name": {
+                        "description" : "Name of the JavaScript class holding the model implementation",
+                        "type": "string"
+                    },
+                    "model_module": {
+                        "description" : "Name of the JavaScript module holding the model implementation",
+                        "type": "string"
+                    },
+                    "model_module_version": {
+                        "description" : "Semver range for the JavaScript module holding the model implementation",
+                        "type": "string"
+                    },
+                    "state": {
+                        "description" : "Serialized state of the model",
+                        "type": "object"
+                    },
+                    "buffer_paths": {
+                        "description" : "Array of paths in the state for the corresponding buffers",
+                        "type": "array",
+                        "items": {
+                            "description": "A path for a binary buffer value.",
+                            "type": "array",
+                            "items": {
+                                "description": "An object key or array index",
+                                "anyOf": [{"type": "string"}, {"type": "number"}]
+                            }
+                        }
+                    },
+                    "buffers": {
+                        "description" : "Array of base64-encoded binary buffers",
+                        "type": "array",
+                        "items": {
+                            "description": "A base64-encoded binary buffer",
+                            "type": "string"
+                        }
```

Comment:
I'd say, include the encoding(s) in the description, since if at some moments we decide to support compression, we may have to change the schema again. If we put the encoding in the state file it's more flexible.